src/services/counters/CounterService.ts

```diff
@@ -0,0 +1,176 @@
+import {isNotNullOrUndefined, traverse, deepValue} from "../../common/utils/Utils";
+import {ICacheContainer} from "../container/ICacheContainer";
+import {CompositeOpBuilder} from "../dmr/CompositeOpBuilder";
+import {IDmrCompositeReq} from "../dmr/IDmrCompositeReq";
+import {JGroupsService} from "../jgroups/JGroupsService";
+import {IServerAddress} from "../server/IServerAddress";
+import {WeakCounter} from "./WeakCounter";
+import {StrongCounter} from "./StrongCounter";
+
+const module: ng.IModule = App.module("managementConsole.services.counter", []);
+
+export class CounterService {
+  static $inject: string[] = ["$q", "jGroupsService", "dmrService", "socketBindingService", "launchType"];
+
+  constructor(private $q: IQService, private jgroupsService: JGroupsService, private dmrService: DmrService,
+              private socketBindingService: SocketBindingService, private launchType: LaunchTypeService) {
+  }
+
+  createStrongCounter(name: string, storage: string, initialValue: number, currentValue: number, lowerBound:number, upperBound: number, dmr: any): ICounter {
+    return new StrongCounter(name, storage, initialValue, currentValue, lowerBound, upperBound, dmr);
+  }
+
+  createWeakCounter(name: string, storage: string, initialValue: number, currentValue: number, concurrency: number, dmr: any): ICounter {
+    return new WeakCounter(name, storage, initialValue, currentValue, concurrency, dmr);
+  }
+
+  doCountersExist(cacheContainer: string, profile: string): ng.IPromise<boolean> {
+    let deferred: ng.IDeferred<boolean> = this.$q.defer<boolean>();
+    let counterAddress: string [] = this.getCountersConfigurationAddress(cacheContainer, profile);
+    let request: IDmrRequest = {
+      address: counterAddress
+    };
+    this.dmrService.readResource(request).then(r => {
+      deferred.resolve(true);
+    }).catch(e => {
+      deferred.resolve(false);
+    });
+    return deferred.promise;
+  }
+
+  createCountersDMRPath(cacheContainer: string, profile: string): ng.IPromise<boolean> {
+    let deferred: ng.IDeferred<boolean> = this.$q.defer<boolean>();
+    let counterAddress: string [] = this.getCountersConfigurationAddress(cacheContainer, profile);
+    let request: IDmrRequest = {
+      address: counterAddress
+    };
+    this.dmrService.add(request).then(r => {
+      deferred.resolve(true);
+    }).catch(e => {
+      deferred.resolve(false);
+    });
+    return deferred.promise;
+  }
+
+  getAllCounters(cacheContainer: ICacheContainer): ng.IPromise<ICounter[]> {
+
+    let deferred: ng.IDeferred<ICounter[]> = this.$q.defer<ICounter[]>();
+    this.jgroupsService.getServerGroupCoordinator(cacheContainer.serverGroup).then((coord: IServerAddress) => {
+
+      let request: IDmrRequest = <IDmrRequest>{
+        address: this.getCountersRuntimeAddress(cacheContainer.name, coord.host, coord.name),
+        "recursive-depth": 2,
+        "include-runtime": true
+      };
+
+      this.dmrService.readResource(request)
```

Comment:
Will this not fail if `counters=COUNTERS` hasn't been created?

Reply:
^ I guess not because it's guarded in `CacheContainer.ts` but we should really have this check in the service